### PR TITLE
filter device parameters based on arn

### PR DESCRIPTION
**Pull request by ajberdy**
*Issue #, if available:*

*Description of changes:*

Filter parameters property of BraketSampler and BraketDWaveSampler based on the deviceParameters for the specific qpu being used.

*Testing done:*

Added unit tests

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [X] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-ocean-plugin-python/blob/main/CONTRIBUTING.md) doc
- [X] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-ocean-plugin-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [X] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-ocean-plugin-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-ocean-plugin-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [X] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [X] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/unit_tests/braket/ocean_plugin/conftest.py [hunk 1]
```diff
@@ -67,7 +66,219 @@ def provider_properties():
 
 
 @pytest.fixture
-def braket_sampler_properties(provider_properties, service_properties):
+def two_thousand_q_device_parameters():
```
Comment (jheckey) [inline test/unit_tests/braket/ocean_plugin/conftest.py:69]: This should be loaded from the schema library

File: test/unit_tests/braket/ocean_plugin/conftest.py [hunk 2]
```diff
@@ -67,7 +66,219 @@ def provider_properties():
...
+                        "type": ["integer", "null"],
+                    },
+                }
+            }
+        },
+    }
+
+
+@pytest.fixture
+def advantage_device_parameters():
```
Comment (jheckey) [inline test/unit_tests/braket/ocean_plugin/conftest.py:164]: This should be loaded from the schema library

File: test/unit_tests/braket/ocean_plugin/test_braket_sampler.py [hunk 3]
```diff
@@ -62,6 +72,18 @@ def test_parameters(braket_sampler):
     assert isinstance(braket_sampler.parameters, FrozenDict)
 
 
+def test_advantage_parameters(advantage_braket_sampler):
+    expected_params = {
+        param: ["parameters"] for param in BraketSolverMetadata.DWAVE["parameters"].values()
+    }
+    del expected_params["beta"]
+    del expected_params["chains"]
+    del expected_params["postprocessingType"]
```
Comment (jheckey) [inline test/unit_tests/braket/ocean_plugin/test_braket_sampler.py:81]: Why do these need to be explicitly deleted? They shouldn't be present in the first place.

Comment (ajberdy) [reply test/unit_tests/braket/ocean_plugin/test_braket_sampler.py:81]: They are deleted from `expected_params` which is constructed from the list of parameters in the `BraketSolverMetadata` class

**Review by speller26 — APPROVED** [review]

**Pull request merged**